app/Jobs/Invoice/CreateXInvoice.php

```diff
@@ -0,0 +1,388 @@
+<?php
+
+namespace App\Jobs\Invoice;
+
+use App\Models\Invoice;
+use App\Models\Country;
+use horstoeko\zugferd\ZugferdDocumentBuilder;
+use horstoeko\zugferd\ZugferdDocumentPdfBuilder;
+use horstoeko\zugferd\ZugferdProfiles;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Storage;
+
+
+class CreateXInvoice implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    public Invoice $invoice;
+
+    public function __construct(Invoice $invoice)
+    {
+        $this->invoice = $invoice;
+    }
+
+    /**
+     * Execute the job.
+     *
+     *
+     * @return string
+     */
+    public function handle(): string
+    {
+        $invoice = $this->invoice;
+        $company = $invoice->company;
+        $client = $invoice->client;
+        $profile = "";
+        switch ($company->xinvoice_type){
+            case "EN16931":
+                $profile = ZugferdProfiles::PROFILE_EN16931;
+                break;
+            case "XInvoice_2_2":
+                $profile = ZugferdProfiles::PROFILE_XRECHNUNG_2_2;
+                break;
+            case "XInvoice_2_1":
+                $profile = ZugferdProfiles::PROFILE_XRECHNUNG_2_1;
+                break;
+            case "XInvoice_2_0":
+                $profile = ZugferdProfiles::PROFILE_XRECHNUNG_2;
+                break;
+            case "XInvoice_1_0":
+                $profile = ZugferdProfiles::PROFILE_XRECHNUNG;
+                break;
+            case "XInvoice-Extended":
+                $profile = ZugferdProfiles::PROFILE_EXTENDED;
+                break;
+            case "XInvoice-BasicWL":
+                $profile = ZugferdProfiles::PROFILE_BASICWL;
+                break;
+            case "XInvoice-Basic":
+                $profile = ZugferdProfiles::PROFILE_BASIC;
+                break;
+        }
+        $xrechnung =  ZugferdDocumentBuilder::CreateNew($profile);
+
+        $xrechnung
+            ->setDocumentInformation($invoice->number, "380", date_create($invoice->date), $invoice->client->getCurrencyCode())
+            ->addDocumentNote($invoice->public_notes)
+            ->setDocumentSupplyChainEvent(date_create($invoice->date))
+            ->setDocumentSeller($company->getSetting('name'))
+            ->setDocumentSellerAddress($company->getSetting("address1"), "", "", $company->getSetting("postal_code"), $company->getSetting("city"), "Germany") //Country::query()->where('id', $company->getSetting("country_id"))->first()
```

Comment:
@turbo124 how can I get the company's country here in iso_3166_2?

Reply:
$company->country()->iso_3166_2